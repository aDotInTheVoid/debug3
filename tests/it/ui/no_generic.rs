use debug3::{pprint, Debug};

#[derive(Debug)]
struct Triple<X, Y, Z> {
    x: X,
    y: Y,
    z: Z,
}

struct NoDebug;

fn main() {
    pprint(Triple {
        x: 1u8,
        y: 2u16,
        z: NoDebug,
    });
}
