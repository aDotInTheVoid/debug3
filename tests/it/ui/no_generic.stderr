error[E0277]: the trait bound `NoDebug: debug3::Debug` is not satisfied
  --> tests/it/ui/no_generic.rs:16:12
   |
13 |     pprint(Triple {
   |     ------ required by a bound introduced by this call
...
16 |         z: NoDebug,
   |            ^^^^^^^ the trait `debug3::Debug` is not implemented for `NoDebug`
   |
   = help: the following other types implement trait `debug3::Debug`:
             bool
             char
             isize
             i8
             i16
             i32
             i64
             i128
           and $N others
note: required for `Triple<u8, u16, NoDebug>` to implement `debug3::Debug`
  --> tests/it/ui/no_generic.rs:3:10
   |
3  | #[derive(Debug)]
   |          ^^^^^ unsatisfied trait bound introduced in this `derive` macro
4  | struct Triple<X, Y, Z> {
   |        ^^^^^^^^^^^^^^^
note: required by a bound in `pprint`
  --> src/lib.rs
   |
   | pub fn pprint<T: Debug>(x: T) -> String {
   |                  ^^^^^ required by this bound in `pprint`
   = note: this error originates in the derive macro `Debug` (in Nightly builds, run with -Z macro-backtrace for more info)
